src/Microsoft.Diagnostics.Monitoring.WebApi/Stacks/NameCache.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Concurrent;
+using System.Diagnostics;
+
+#if STARTUPHOOK
+namespace Microsoft.Diagnostics.Monitoring.StartupHook
```

Comment:
Are you avoiding dependencies for the startup hook dll?

Reply:
Yes, I'm trying to put as little code (within reason) into any assembly that we want to load into the target application.